internal/oci/container.go

```diff
@@ -319,8 +321,8 @@ func (c *Container) Created() bool {
 
 // SetStartFailed sets the container state appropriately after a start failure
 func (c *Container) SetStartFailed(err error) {
-	c.opLock.Lock()
-	defer c.opLock.Unlock()
+	c.stateLock.Lock()
```

Comment:
I expeced more instances of the write lock in use.

Reply:
Container#FromDisk should use write lock as well.